web/src/lib/auth/sessionHelper.ts

```diff
@@ -93,21 +107,26 @@ export const getSignedS3Link = async (value: string, expires?: number): Promise<
   return formatUrl(url);
 };
 
-export const getCurrentToken = async (): Promise<string> => {
-  const cognitoSession = await Auth.currentSession();
-  return cognitoSession.getIdToken().getJwtToken();
+export const getCurrentToken = async (): Promise<JWT> => {
+  const session = await fetchAuthSession({ forceRefresh: true });
+  if (!session || !session.tokens || !session.tokens.idToken) {
```

Comment:
[dust] Could we simplify this?
```suggestion
  if (!session.tokens.idToken) {
```

Reply:
I've simplified it by removing the redundant !session check since fetchAuthSession is expected to always return a valid session. The updated code focuses on verifying session.tokens and session.tokens.idToken, improving clarity and readability while ensuring TypeScript safety.